base/acme/src/main/java/org/dogtagpki/acme/issuer/PKIIssuer.java

```diff
@@ -158,15 +159,19 @@ public String issueCertificate(String csr) throws Exception {
             throw new Exception("Unable to generate certificate: " + error);
         }
 
-        CertReviewResponse reviewInfo = certClient.reviewRequest(requestId);
-        certClient.approveRequest(requestId, reviewInfo);
+        CertId id = null;
+        if (info.getRequestStatus() == RequestStatus.COMPLETE) {
+            id = info.getCertId();
+        } else {
+            CertReviewResponse reviewInfo = certClient.reviewRequest(requestId);
+            certClient.approveRequest(requestId, reviewInfo);
 
-        info = certClient.getRequest(requestId);
-        logger.info("Serial number: " + info.getCertId().toHexString());
+            info = certClient.getRequest(requestId);
+            id = info.getCertId();
```

Comment:
This statement and the one in line 164 could be moved after the if-else statement, but it's not a big deal.